src/flowchem/server/fastapi_server.py

```diff
@@ -39,13 +39,12 @@ def home_redirect_to_docs(request):
 
     def add_background_tasks(self, repeated_tasks: Iterable[RepeatedTaskInfo]):
         """Schedule repeated tasks to run upon server startup."""
-        for seconds_delay, task in repeated_tasks:
-
-            @self.app.on_event("startup")
-            @repeat_every(seconds=seconds_delay)
-            async def my_task():
-                logger.debug("Running repeated task...")
-                await task()
+        seconds_delay, task = repeated_tasks
```

Comment:
this changes behaviour and is not in line with type annotation. check, explain and fix